### ISSUE TITLE
MPasswordField not adjusting save button state of AbstractForm properly

### ISSUE THREAD

**Issue by sischi** [viritin/viritin#272]
Hi Matti,

it seems there is an usability-related issue with the MPasswordField in combination with adjusting the save button state of the AbstractForm.

Lets take a look at the [MWE](https://github.com/viritin/viritin/files/661314/mpasswordfield_demo.txt) attached.

There we have a simple LoginForm that extends AbstractForm with a MTextField to input an username and a MPasswordField to obviously input the corresponding password. The backed bean of type LoginModel defines two Strings that are @NotNull annotated.
So now you focus the username textfield to enter a username and then switch to the passwordfield to enter a password. I would expect, that while typing any password, the save button will be enabled, because both fields are filled up and not null anymore. Instead the save button remains disabled and the user thinks he has done something wrong UNTIL the passwordfield loses its focus.

If you use a MTextField instead of a MPasswordField for entering the password, enabling the save button works as expected.

A quick look into the validation process of the MPasswordField shows, that the value of the backed bean seems not beeing updated until the field loses focus. So the validation process of the MBeanFieldGroup (the isValid() method) fails at the jsr303ValidateBean(...) method, which validates the bean instead of the current field values like the rest of the isValidAllProperties() method.

Compared to the MPasswordField, the MTextField updates the bean value in the doEagerValidation() method:
``` Java
    // MTextField
    protected void doEagerValidation() {
        final boolean wasvalid = eagerValidationStatus;
        eagerValidationStatus = true;
        eagerValidationError = null;
        try {
            if (isRequired() && getLastKnownTextContent().isEmpty()) {
                throw new Validator.EmptyValueException(getRequiredError());
            }
            validate(getLastKnownTextContent());
            if (!wasvalid) {
                markAsDirty();
            }
            // Also eagerly pass content to backing bean to make top level
            // validation eager, but do not listen the value back in value change
            // event
            if (getPropertyDataSource() != null) {
                skipValueChangeEvent = true;
                Object convertedValue = ConverterUtil.convertToModel(
                        getLastKnownTextContent(), getPropertyDataSource().
                        getType(), getConverter(),
                        getLocale());
                getPropertyDataSource().setValue(convertedValue);
                skipValueChangeEvent = false;
            }
        } catch (Validator.InvalidValueException e) {
            eagerValidationError = e;
            eagerValidationStatus = false;
            markAsDirty();
        }
    }

    // MPasswordField
    protected void doEagerValidation() {
        final boolean wasvalid = eagerValidationStatus;
        eagerValidationStatus = true;
        eagerValidationError = null;
        try {
            if (isRequired() && getLastKnownTextContent().isEmpty()) {
                throw new Validator.EmptyValueException(getRequiredError());
            }
            validate(getLastKnownTextContent());
            if (!wasvalid) {
                markAsDirty();
            }
        } catch (Validator.InvalidValueException e) {
            eagerValidationError = e;
            eagerValidationStatus = false;
            markAsDirty();
        }
    }
```

Is there any reason for the different behaviour of those two fields?

Thanks in advance.

**Comment by mstahv** [comment]
Nope, I think they should be the same. I guess before Java 8 we'll just have to accept some duplication.

Want to create a pull request? The usage example could also be pushed as a test code.

**Comment by sischi** [comment]
While preparing a pull request, I recognized that this issue was "fixed" in the most recent release (1.61). It seems to be fixed since build 1.59 (I was using 1.58).

The commit, that resolved this issue was 3d7cfc4:
``` Java
if (!containsAtLeastOneBoundComponentWithError && isValidateOnlyBoundFields()) {
    return true;
}
```
Because of this statement at the end of the jsr303ValidateBean(...) routine in MBeanFieldGroup, the fieldgroup becomes valid and the save button will be enabled.
